From e3b825a72037538bdb7b181bb39cabc3f121d0d4 Mon Sep 17 00:00:00 2001
From: Andrew Beekhof <andrew@beekhof.net>
Date: Sun, 29 Oct 2017 16:33:13 +1100
Subject: [PATCH] Fix: crm_resource: Ensure we wait for all messages before
 exiting

- Wait for deletions from remote nodes too

Terminating too early prevents our delete operations being read by the
crmd process.
---
 tools/crm_resource_runtime.c | 2 --
 1 files changed, 0 insertions(+), 2 deletions(-)

diff --git a/tools/crm_resource_runtime.c b/tools/crm_resource_runtime.c
index cab920b..809f641 100644
--- a/tools/crm_resource_runtime.c
+++ b/tools/crm_resource_runtime.c
@@ -661,9 +661,7 @@ cli_resource_delete(crm_ipc_t *crmd_channel, const char *host_uname,
     if (rc == pcmk_ok) {
         char *attr_name = NULL;
 
-        if(node && node->details->remote_rsc == NULL && node->details->rsc_discovery_enabled) {
             crmd_replies_needed++;
-        }
 
         if(is_not_set(rsc->flags, pe_rsc_unique)) {
             char *id = clone_strip(rsc->id);
-- 
1.8.3.1

